api/base_controller.py

```diff
@@ -85,9 +75,21 @@ def authenticated_patron(self, authorization_header):
 
     def authenticate(self):
         """Sends a 401 response that demands authentication."""
-        if not self.manager.opds_authentication_document:
-            self.manager.opds_authentication_document = self.manager.auth.create_authentication_document()
+        library_short_name = flask.request.library.short_name
+        if library_short_name not in self.manager.opds_authentication_documents:
+            self.manager.opds_authentication_documents[library_short_name] = self.manager.auth.create_authentication_document()
 
-        data = self.manager.opds_authentication_document
+        data = self.manager.opds_authentication_documents[library_short_name]
         headers = self.manager.auth.create_authentication_headers()
         return Response(data, 401, headers)
+
+    def library_from_request(self, library_short_name):
```

Comment:
Maybe I should rename this to library_from_short_name. It's not much like authenticated_patron_from_request.